### DISCUSSION TITLE
Proper Update Method for Windows

### DISCUSSION THREAD

**Discussion by pavichokche** [rogerfar/rdt-client#564] [Q&A] [answered]
I'm running the client as a service on Windows. Every time I update I shut it down, back up the program directory, and then extract the new release zip file EXCEPT the web.config and appsettings.* files. Is that correct? Is there a proper update methodology for Windows?

**Comment by mtechnomaniac** [comment]
> I'm running the client as a service on Windows. Every time I update I shut it down, back up the program directory, and then extract the new release zip file EXCEPT the web.config and appsettings.* files. Is that correct? Is there a proper update methodology for Windows?

I'm also looking for the right instructions to update. please confirm if this is the best way to do it

**Comment by mtechnomaniac** [comment]
> I'm running the client as a service on Windows. Every time I update I shut it down, back up the program directory, and then extract the new release zip file EXCEPT the web.config and appsettings.* files. Is that correct? Is there a proper update methodology for Windows?

I'm not sure if you found the correct solution yet but here's what I did instead of manually downloading the udpate:
1. Open 'Windows Powershell' with admin
2. Navigate to the folder where RDT client is installed
3. Type the following command (without quotes) for autoupdate: ".\update.ps1"

I initially got the following error: "ps1 cannot be loaded because running scripts is disabled on this system. For more information, see about_Execution_Policies at https:/go.microsoft.com/fwlink/?LinkID=135170"
Used this command to enable scripts in PowerShell: "Set-ExecutionPolicy -ExecutionPolicy Unrestricted"
You can confirm if the policy is enabled using: "Get-ExecutionPolicy" 
Output should be "Unrestricted"
Additionally I disabled 'running scripts' using the following command (of course after the update completed sucessfully): "Set-ExecutionPolicy -ExecutionPolicy Restricted"

P.S: There is no need to backup files before updating RDT Client but I did it nevertheless and made a copy of the entire folder (old version), in case something goes wrong and I needed a fallback

**Reply by medinauta** [reply]
There's a file called How to update in the root folder where you need to open a power shell with admin rights and run ".\update.ps1". Unfortunately for me it didn't work. After downloading and extracting the files I got an error that it couldn't replace a directory that already existed (the root). After a little reading, I just gave up and did it manually, extracted the zip and replace files. The most important files are the appsettings and the database (which is unaffected for the files replacement).

**Reply by mtechnomaniac** [reply]
We are trying to update on Windows and can't find any 'How to Update' file in root folder. There's only that 'update.ps1', at least in versions 2.0.81 & above

**Reply by pavichokche** [reply]
> The most important files are the appsettings and the database (which is unaffected for the files replacement).

The release .zip being distributed DOES contain an appsettings.json file, so our most important files would be affected by doing a straight-forward file replacement...

**Reply by flatbr0ke** [reply, marked as answer]
> There's a file called How to update in the root folder where you need to open a power shell with admin rights and run ".\update.ps1". Unfortunately for me it didn't work. After downloading and extracting the files I got an error that it couldn't replace a directory that already existed (the root). After a little reading, I just gave up and did it manually, extracted the zip and replace files. The most important files are the appsettings and the database (which is unaffected for the files replacement).

I had the same issue getting error  "couldn't replace directory that already existed", but I finally found a solution.
1. Goto directory that RDB Client is in and right click on update.ps1 file. select properties>security.
2. Under group or user names, make sure that all have full control selected. If not, select edit and allow it then select OK.
3. open PowerShell in the same directory that RDB Client is in, giving it Admin rights by holding shift and right clicking to select "Open PowerShell Window Here"
4. type in ".\update.ps1" without quotes and enter
5. should stop your service and download the update to install . ( I did receive an error about the directory), but this time it finished and closed the PowerShell box.
6. open RDB Client in your browser and refresh to check your version (Mine was finally updated to the newest version.)

Hope this helps anyone having this issue!

**Reply by whatmeworry68** [reply]
> > There's a file called How to update in the root folder where you need to open a power shell with admin rights and run ".\update.ps1". Unfortunately for me it didn't work. After downloading and extracting the files I got an error that it couldn't replace a directory that already existed (the root). After a little reading, I just gave up and did it manually, extracted the zip and replace files. The most important files are the appsettings and the database (which is unaffected for the files replacement).
> 
> I had the same issue getting error "couldn't replace directory that already existed", but I finally found a solution.
> 
>     1. Goto directory that RDB Client is in and right click on update.ps1 file. select properties>security.
> 
>     2. Under group or user names, make sure that all have full control selected. If not, select edit and allow it then select OK.
> 
>     3. open PowerShell in the same directory that RDB Client is in, giving it Admin rights by holding shift and right clicking to select "Open PowerShell Window Here"
> 
>     4. type in ".\update.ps1" without quotes and enter
> 
>     5. should stop your service and download the update to install . ( I did receive an error about the directory), but this time it finished and closed the PowerShell box.
> 
>     6. open RDB Client in your browser and refresh to check your version (Mine was finally updated to the newest version.)
> 
> 
> Hope this helps anyone having this issue!

I was finally able to test this and it worked just as you described.  Thank you!

**Reply by pavichokche** [reply]
I confirmed that the solution @flatbr0ke provided above works!
I also got a scary-looking error message about files access, but the program was successfully updated without me having to juggle around zip files and carefully overwrite files.
